OMCompiler/Compiler/NFFrontEnd/NFFlatten.mo

```diff
@@ -741,6 +749,58 @@ algorithm
   isNamed := name == attr_name;
 end isTypeAttributeNamed;
 
+function verifyBinding
+  input Variable var;
+  input Variability variability;
+  input Binding binding;
+protected
+  Binding fixed_binding;
+  Expression fixed_exp;
+  Boolean fixed;
+algorithm
+  if variability > Variability.CONSTANT and Binding.isBound(binding) then
+    // Parameter with a binding is ok.
+    return;
+  end if;
+
+  // Check if the variable is fixed or not.
+  fixed_binding := Variable.lookupTypeAttribute("fixed", var);
+
+  if Binding.isBound(fixed_binding) then
+    fixed_exp := Binding.getExp(fixed_binding);
+    fixed_exp := Ceval.tryEvalExp(fixed_exp);
+
+    if not Expression.isBoolean(fixed_exp) then
+      return;
+    end if;
+
+    fixed := Expression.isTrue(fixed_exp);
+  else
+    fixed := true;
+  end if;
+
+  if variability == Variability.CONSTANT then
+    if not fixed then
+      // Constants are not allowed to be non-fixed.
+      Error.addSourceMessage(Error.NON_FIXED_CONSTANT,
+        {ComponentRef.toString(var.name)}, var.info);
+      fail();
+    end if;
+
+    // Constants also must have binding equations if they are used, but this is
+    // checked when evaluating them.
+
+  else
+    if fixed and Binding.isUnbound(binding) and
+         Binding.isUnbound(Variable.lookupTypeAttribute("start", var)) then
```

Comment:
Can you give a warning if it is unbound but start attribute does exist?

Reply:
Done